AdsiPS/Public/New-ADSIUser.ps1

```diff
@@ -147,7 +147,7 @@
 				IF ($PSBoundParameters['HomeDrive']) { $user.HomeDrive = $HomeDrive }
 				IF ($PSBoundParameters['MiddleName']) { $user.MiddleName = $MiddleName }
 				IF ($PSBoundParameters['VoiceTelephoneNumber']) { $user.VoiceTelephoneNumber }
-				IF ($PSBoundParameters['AccountPassword']){$User.SetPassword($AccountPassword)}
+				IF ($PSBoundParameters['AccountPassword']){$User.SetPassword((New-Object PSCredential "user",$AccountPassword).GetNetworkCredential().Password)}
```

Comment:
quick question on this, is there a reason why [System.Security.SecureString]$AccountPassword would not do the job ? Is there an advantage to create a new PSCredential object ?

Reply:
[SetPassword](https://msdn.microsoft.com/en-us/library/system.directoryservices.accountmanagement.authenticableprincipal.setpassword(v=vs.110).aspx) only takes a password in plaintext, so using [System.Security.SecureString]$AccountPassword was actually setting the password to "System.Security.SecureString".

Creating the PSCredential object is just an easy way to convert the SecureString to a String.

Reply:
Alright sounds good, thanks again @andrewtchilds ! Really appreciate your contribution!